src/main/java/com/synopsys/integration/detect/lifecycle/boot/DetectBootFactory.java

```diff
@@ -48,10 +141,96 @@ public Configuration createConfiguration() {
 
     public DocumentBuilder createXmlDocumentBuilder() {
         try {
-            final DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
             return factory.newDocumentBuilder();
-        } catch (final Exception e) {
+        } catch (Exception e) {
             throw new RuntimeException(e);
         }
     }
+
+    public PathResolver createPathResolver(Boolean resolveTildes) {
+        PathResolver pathResolver;
+
+        if (detectInfo.getCurrentOs() != OperatingSystemType.WINDOWS && Boolean.TRUE.equals(resolveTildes)) {
+            logger.info("Tilde's will be automatically resolved to USER HOME.");
+            pathResolver = new TildeInPathResolver(SystemUtils.USER_HOME);
+        } else {
+            pathResolver = new SimplePathResolver();
+        }
+
+        return pathResolver;
+    }
+
+    public Optional<DiagnosticSystem> createDiagnosticSystem(DiagnosticsDecider diagnosticsDecider, PropertyConfiguration detectConfiguration, DirectoryManager directoryManager) {
```

Comment:
I'd prefer the logic for isConfigured be done in boot and this just does the creation (though it may then make less sense if that is all it does). But other than this there seems to be no logic (or only very trivial logic) in the factory, and this seems more appropriate for the manager.

Reply:
I've moved isConfigured logic into the manager, seems like it's better to be done there than in either DetectBoot or the Factory.